examples/experimental/imagenet_gaea/model_def.py

```diff
@@ -0,0 +1,203 @@
+            bucket_name=bucket_name,
+            transform=
+            transforms.Compose(
+                [
+                    transforms.Resize(256),
+                    transforms.CenterCrop(224),
+                    transforms.ToTensor(),
+                    normalize,
+                ]
+            ),
+        )
+
+        valid_queue = DataLoader(
+            valid_data,
+            batch_size=self.context.get_per_slot_batch_size(),
+            shuffle=False,
+            pin_memory=False,
+            num_workers=2,
+        )
+        return valid_queue
+
+    def build_model(self) -> nn.Module:
+        genotype = Genotype(
+            normal=[
+                ('skip_connect', 1),
+                ('skip_connect', 0),
+                ('sep_conv_3x3', 2),
+                ('sep_conv_3x3', 1),
+                ('sep_conv_5x5', 2),
+                ('sep_conv_3x3', 0),
+                ('sep_conv_5x5', 3),
+                ('sep_conv_5x5', 2)
+            ], normal_concat=range(2,6),
+            reduce=[
+                ('max_pool_3x3', 1),
+                ('sep_conv_3x3', 0),
+                ('sep_conv_5x5', 1),
+                ('dil_conv_5x5', 2),
+                ('sep_conv_3x3', 1),
+                ('sep_conv_3x3', 3),
+                ('sep_conv_5x5', 1),
+                ('max_pool_3x3', 2)
+            ], reduce_concat=range(2,6)
+        )
+        activation_function = activation_map[self.context.get_hparam("activation")]
+
+        model = NetworkImageNet(
+            genotype,
+            activation_function,
+            self.context.get_hparam("init_channels"),
+            1000, # number of classes
+            self.context.get_hparam("layers"),
+            auxiliary=self.context.get_hparam("auxiliary"),
+            do_SE=self.context.get_hparam("do_SE")
+        )
+
+        # If loading backbone weights, do not call reset_parameters() or
+        # call before loading the backbone weights.
+        reset_parameters(model)
+        return model
+
+    def optimizer(self, model: nn.Module) -> torch.optim.Optimizer:  # type: ignore
+        self._optimizer = torch.optim.SGD(
+            model.parameters(),
+            lr=self.context.get_hparam("learning_rate"),
+            momentum=self.context.get_hparam("momentum"),
+            weight_decay=self.context.get_hparam("weight_decay")
+        )
+        return self._optimizer
+
+    def create_lr_scheduler(self, optimizer):
+        self.scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(
+            optimizer, 300
+        )
+        step_mode = LRScheduler.StepMode.MANUAL_STEP
+        return LRScheduler(self.scheduler, step_mode=step_mode)
+
+    def train_batch(
+        self, batch: Any, model: nn.Module, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, torch.Tensor]:
+
+        if batch_idx==0 or self.last_epoch_idx < epoch_idx:
+            self.scheduler.step()
+            current_lr = self.scheduler.get_lr()[0]
```

Comment:
This LR scheduler should have a function called get_last_lr() which pulls the correct lr vs get_lr() which can also calculate the LRs.

Reply:
Thanks for pointing that out.  Changed to get_last_lr.